broker/amqp.go

```diff
@@ -52,22 +54,27 @@ func (a *AMQP) Connect(url string) error {
 	return err
 }
 
-// Close implements io.Closer and Closes the Channel & Connection of this Client
+// Close implements io.Closer and closes the publishing Channel & Connection of this Client
 func (a *AMQP) Close() error {
-	err := a.channel.Close()
+	err := a.publishChannel.Close()
 	if err != nil {
 		return err
 	}
-	err = a.conn.Close()
+	err = a.publishConn.Close()
 	if err != nil {
 		return err
 	}
+	if a.consumeConn != nil {
+		if err := a.consumeConn.Close(); err != nil {
+            		return err
```

Comment:
This can be simplified to just `return a.consumeConn.Close()`

Reply:
:+1: